src/sentry/utils/appleconnect/itunes_connect.py

```diff
@@ -302,9 +302,26 @@ def _request_trusted_phone_info(self) -> None:
             },
             timeout=REQUEST_TIMEOUT,
         )
-        if response.status_code != HTTPStatus.OK:
-            raise ITunesError(f"Unexpected response status: {response.status_code}")
-        info = response.json()["trustedPhoneNumber"]
+        # Fastlane actually doesn't check the response status in their implementation and simply
+        # checks whether the response returns the expected fields associated with 2FA. This is
+        # tries to be lenient in a similar way.
+        is_expected_status = (
+            response.status_code == HTTPStatus.OK or response.status_code == HTTPStatus.CREATED
+        )
+        try:
+            info = response.json()["trustedPhoneNumber"]
+        except KeyError:
```

Comment:
If the response does not contain JSON, this will throw `requests.exceptions.JSONDecodeError`, which we should also catch.

Reply:
for posterity: since we're only checking for `response.ok` it's still possible to get a `JSONDecodeError` if we receive a 204 No Content.

Reply:
catching `ValueError` instead because apparently `requests.exceptions` no longer has `JSONDecodeError`, and `json()`'s docstring says it throws `ValueError` if something's wrong with the JSON